pandas/_libs/index.pyx

```diff
@@ -1220,6 +1220,7 @@ cdef class MaskedIndexEngine(IndexEngine):
                             n_alloc *= 2
                             if n_alloc > max_alloc:
                                 n_alloc = max_alloc
+                            result = np.resize(result, n_alloc)
```

Comment:
Is this required to resize here? If so can we get rid of the resize that happens after this branch is completed? Might be misreading but it looks duplicative

Reply:
I think this is required. The result size isn't known so I believe the resize check needs to be done each time a new value is set. I don't see a way to avoid it with the current algo.

Reply:
Ah yea that makes sense. I think its worth making the conditional resize its own function then so we don't have as much duplication

Reply:
good idea - done